### ISSUE TITLE
Transparent background on desktop but not when a window is open

### ISSUE THREAD

**Issue by gamehelp16** [micheleg/dash-to-dock#375]
It would be nice if we can make it so that the dock background is transparent on the desktop when the dock doesn't auto hide and give it an opaque background when there's a window open and the dock auto-hides.

**Comment by corebots** [comment]
this would be a nice option to have indeed;

**Comment by gamehelp16** [comment]
Looks like there's already a PR about this a few years ago.. https://github.com/micheleg/dash-to-dock/pull/62

**Comment by peterychuang** [comment]
I was hoping to have this feature as well.

**Comment by bluefrg** [comment]
This would compliment [dynamic panel transparency](https://extensions.gnome.org/extension/1011/dynamic-panel-transparency/) extension really well if this was integrated.

**Comment by franglais125** [comment]
Something like this was implemented upstream, in the gnome-shell source code, in preparation for Gnome 3.26.

It would be neat if we could 'blend in' and implement this as well, so that the behavior is more uniform.

Here is the upstream commit:
https://git.gnome.org/browse/gnome-shell/commit/?id=447bf55e45b00426ed908b1b1035f472c2466956

**Comment by micheleg** [comment]
Yes.I agree, especially in side/panel mode, the mismatch between the transparent panel and the opaque dock is quite ugly.

**Comment by franglais125** [comment]
Here is some tentative work on this: https://github.com/franglais125/dash-to-dock/commits/transparency

I still have to add a an option to the settings UI, but the wiring is done. It seems to work ok, though more testing is certainly needed.
